lib/node_modules/@stdlib/lapack/base/dlange/lib/base.js

```diff
@@ -0,0 +1,367 @@
+/**
+* Returns the absolute value of the frobenius norm of a real matrix `A`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @returns {number} result
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+*
+* var result = frobeniusNorm( 3, 4, A, 4, 1, 0 );
+* // returns ~25.5
+*/
+function frobeniusNorm( M, N, A, strideA1, strideA2, offsetA ) {
+	var da0;
+	var da1;
+	var S1;
+	var S2;
+	var ia;
+	var i;
+
+	workspace[ 0 ] = 0.0;
+	workspace[ 1 ] = 1.0;
+
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		S1 = M;
+		S2 = N;
+		da0 = strideA2;
+		da1 = strideA1;
+	} else {
+		S1 = N;
+		S2 = M;
+		da0 = strideA1;
+		da1 = strideA2;
+	}
+
+	ia = offsetA;
+	for ( i = 0; i < S1; i++ ) {
+		dlassq( S2, A, da0, ia, workspace[ 0 ], workspace[ 1 ], workspace, 1, 0 ); // eslint-disable-line max-len
+		ia += da1;
+	}
+
+	return workspace[ 0 ] * sqrt( workspace[ 1 ] );
+}
+
+
+// MAIN //
+
+/**
+* Returns the value of the one norm, or the frobenius norm, or the infinity norm, or the element with the largest absolute value of a real matrix `A`.
+*
+* ## Notes
+*
+* -   use `norm` = `max` to calculate the element with the largest absolute value
+* -   use `norm` = `one` to calculate the one norm, work should have `N` indexed elements if row-major layout is used
+* -   use `norm` = `infinity` to calculate the infinity norm, work should have `M` indexed elements if column-major layout is used
+* -   use `norm` = `frobenius` to calculate the frobenius norm
+*
+* @private
+* @param {string} norm - specifies the type of norm to be calculated
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @param {Float64Array} work - temporary workspace array
+* @param {integer} strideWork - stride length of `work`
+* @param {NonNegativeInteger} offsetWork - starting index of `work`
+* @returns {number} result
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+* var work = new Float64Array( 3 );
+*
+* var out = dlange( 'frobenius', 3, 4, A, 4, 1, 0, work, 1, 0 );
+* // returns ~25.5
+*/
+function dlange( norm, M, N, A, strideA1, strideA2, offsetA, work, strideWork, offsetWork ) { // eslint-disable-line max-len
```

Comment:
Rather than disable the lint rule multiple times in this file, just go ahead and disable once above the 'use strict' directive.